Mathlib/Tactic/LibrarySearch.lean

```diff
@@ -92,8 +92,20 @@ def librarySearch (mvarId : MVarId) (lemmas : DiscrTree Name) (solveByElimDepth
 def lines (ls : List MessageData) :=
   MessageData.joinSep ls (MessageData.ofFormat Format.line)
 
+open Lean.Parser.Tactic
+
+-- TODO: implement the additional options for `library_search` from Lean 3,
+-- in particular including additional lemmas
+-- with `library_search [X, Y, Z]` or `library_search with attr`,
+-- or requiring that a particular hypothesis is used in the solution, with `library_search using h`.
+syntax (name := librarySearch') "librarySearch" (" (" &"config" " := " term ")")?
+  (" [" simpArg,* "]")? (" with " (colGt ident)+)? (" using " (colGt ident)+)? : tactic
+
+-- For now we only implement the basic functionality.
+-- The full syntax is recognized, but will produce a "Tactic has not been implemented" error.
+
 open Elab.Tactic Elab Tactic in
-elab tk:"librarySearch" : tactic => do
+elab_rules : tactic | `(tactic| librarySearch%$tk) => do
```

Comment:
Can we just translate the Lean 3 `library_search` to `librarySearch3` instead so that we don't have to match the syntax?

Reply:
Is there a reason that `by librarySearch [args]` shouldn't work in lean 4? I understand that it's not done yet but I want all the lean 3 tactics to have equivalents in lean 4.

Reply:
No that should still work, as well as the `using` argument that is also on the todo list.  The configuration object is almost certainly going to change (the lean 3 one refers to the built-in apply tactic).  We might also want to replace solve-by-elim with Jannis' auto port as discussed on Monday, affecting the `with` syntax.

Most importantly though, aligning these tactics is completely unnecessary.